### ISSUE TITLE
[HELP!!] --data cannot be specified more than once

### ISSUE THREAD

**Issue by wa1618i** [JohnLangford/vowpal_wabbit#1347]
I already converted my numpy data (regression problem) into vw format. 
And the following is the config when I typed in the terminal:

vw --data /home/lemma/VW_model/train.vw --holdout_off -f /home/lemma/VW_model/VW.model -b 25 --loss_function squared --l1 1e-05 --l2 1e-05 --noconstant --bootstrap 8 --bs_type mean --threads 4
using l1 regularization = 1e-05
using l2 regularization = 1e-05
final_regressor = /home/lemma/VW_model/VW.model
Num weight bits = 25
learning rate = 0.5
initial_t = 0
power_t = 0.5

But then it generates error:
Error: option '--data' cannot be specified more than once
finished run
number of examples = 0
weighted example sum = 0.000000
weighted label sum = 0.000000
average loss = -nan
total feature number = 0
vw: option '--data' cannot be specified more than once

**If I don't put --threads 4, the error disappears, meaning I cannot use multicore?
This seems related to this: #1084 
Not sure what's going on here?

**Comment by arielf** [comment]
I can reproduce this 100%.
Workaround: drop the `--threads 4`

**Comment by JohnLangford** [comment]
Why do you want to use --threads?

**Comment by ymarzougui** [comment]
I'm having the same issue with the following command :
`./vowpal_wabbit/vowpalwabbit/vw --oaa 5270 --data train.vw -f binary_model.vw --passes 10 --holdout_after 9000000 -progress 10000`

**Comment by ymarzougui** [comment]
I'm having the same issue with the following command :
`./vowpal_wabbit/vowpalwabbit/vw --oaa 5270 --data train.vw -f binary_model.vw --passes 10 --holdout_after 9000000 -progress 10000`

**Comment by JohnLangford** [comment]
You need to use --progress not -progress. 

I'm closing this for now---"threads" is not really meant to be used on the commandline.  Instead, it was designed for a particular use in a library setting.

**Comment by matanster** [comment]
In case it helps anyone in the future, not to reopen this issue per-se: this error will generally show up when the command being run is malformed. Additional example to the above one: you're building the command in code from variables because you're doing a grid search on training, and you're missing a space preceding a `--`. 

I guess patches for further sanitization of the input arguments are welcome :-)
